### PR TITLE
Testing and docs for List component

### PR DESCRIPTION
**Pull request by mikemai2awesome**
## Jira

http://vjira2:8080/browse/BDS-953

## Summary

1. Added testing for the List component.
2. Fixed an issue with the inline list with inset spacing poking out of its container.
3. Also includes a few other misc fixes required to merge this in and have the build pass
    - Updated the `globby` and `semantic-version` NPM package versions to fix node-glob errors encountered when installing
    - auto-formatted a number of different component JS files based on recently updated Prettier rules (fails the lint checks otherwise)
    - updated the Jest snapshots for the button and icons component to ensure tests pass given some new webkit-prefixed transition styles auto-added via Browserslist

## Details

1. `list.js` in the tests folder now contains tests for all props.
2. `testing.md` now has detailed instructions on manual testing.
3. Certain demo pages were improved to have more clarity.

## How to test

Run the branch locally and read through the testing javascript and markdown.

### CODE REVIEW

File: packages/components/lists/bolt-list/TESTING.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+2. Block display should show a vertical list.
+3. Flex display should show a horizontal list with even width items spanning the full width of the list.
+4. Inline display should show a horizontal list with auto width items.
+5. Inline@breakpoint display should show a vertical list when the browser is smaller than the breakpoint specified, and a horizontal list when the browser is equal to or larger than the breakpoint specified. Breakpoint sizes can be referenced on the [Breakpoint page](https://boltdesignsystem.com/pattern-lab/?p=viewall-visual-styles-breakpoints).
+
+## Scenario: spacing variations
+
+1. Given I am viewing the [spacing variations page](https://boltdesignsystem.com/pattern-lab/patterns/02-components-list-20-list-spacing-variations/02-components-list-20-list-spacing-variations.html).
+2. None spacing should show no spacing in between items.
+3. Xsmall to xlarge spacing should show the perspective spacing in between items. Reference the spacing sizes on the [Spacing page](https://boltdesignsystem.com/pattern-lab/?p=viewall-visual-styles-spacing).
```
Comment (danielamorse) [inline packages/components/lists/bolt-list/TESTING.md]: "should show the respective spacing"

File: packages/components/lists/bolt-list/TESTING.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+1. Given I am viewing the [separator variations page](https://boltdesignsystem.com/pattern-lab/patterns/02-components-list-25-list-separator-variations/02-components-list-25-list-separator-variations.html).
+2. Solid separator should show a solid line in between items.
+3. Dashed separator should show a dashed line in between items.
+
+## Scenario: inset variations
+
+1. Given I am viewing the [inset variations page](https://boltdesignsystem.com/pattern-lab/patterns/02-components-list-30-list-inset-variations/02-components-list-30-list-inset-variations.html).
+2. Inset spacing should show the spacing surrounding the item in all 4 directions (top, right, bottom, left).
+3. Xsmall to xlarge spacing should show the perspective spacing surrounding each item. Reference the spacing sizes on the [Spacing page](https://boltdesignsystem.com/pattern-lab/?p=viewall-visual-styles-spacing).
```
Comment (danielamorse) [inline packages/components/lists/bolt-list/TESTING.md]: ditto

File: docs-site/src/pages/pattern-lab/_patterns/02-components/list/15-list-display-variations.twig [hunk 3]
```diff
@@ -1,25 +1,18 @@
...
+        placeholder_2,
+        placeholder_3,
+      ]
+    } only %}
+  {% endfor %}
 
-<h3>inline@breakpoint</h3>
-<p><mark>This display option means a block list transforming to inline at a specific breakpoint.</mark></p>
+  <h3>Inline@breakpoint</h3>
+  <p><mark>Note: this display option means a block list transforming to inline when the browser is equal to or greater than the breakpoint specified.</mark></p>
```
[the remark this answers is not in the document]
Comment (mikemai2awesome) [reply docs-site/src/pages/pattern-lab/_patterns/02-components/list/15-list-display-variations.twig]: Copy updated.

File: packages/components/lists/bolt-list/TESTING.md (hunk 1 above)
Comment (mikemai2awesome) [reply packages/components/lists/bolt-list/TESTING.md]: Copy updated.

File: packages/components/lists/bolt-list/TESTING.md (hunk 2 above)
Comment (mikemai2awesome) [reply packages/components/lists/bolt-list/TESTING.md]: Copy updated.

**Comment by mikemai2awesome** [comment]
@danielamorse 

I made the copy updates and pulled in master to resolve any conflicts.

**Comment by mikemai2awesome** [comment]
@sghoweri 

I revised the docs based on feedback and added a small CSS fix, which is now indicated in the PR description as well.

**Pull request merged**